### PR TITLE
K8SPS-53 document backups

### PR DESCRIPTION
**Pull request by fiowro**

### CODE REVIEW

File: source/backups.rst [hunk 1]
```diff
@@ -0,0 +1,278 @@
...
+Amazon S3, the `endpointURL <https://docs.min.io/docs/aws-cli-with-minio.html>`_ is needed in the ``s3`` subsection which points to the actual cloud used for backups and
+is specific to the cloud provider. For example, using `Google Cloud <https://cloud.google.com>`_ involves the `following <https://storage.googleapis.com>`_ endpointUrl:
+
+.. code:: yaml
+
+   endpointUrl: https://storage.googleapis.com
+
+Also you can use :ref:`prefix<backup-storages-s3-prefix>` option to specify the
+path (sub-folder) to the backups inside the S3 bucket. If prefix is not set,
+backups are stored in the root directory.
```
Comment (egegunes) [inline source/backups.rst:88]: it's not true in case of PS operator, it'll be implemented in the future

File: source/backups.rst [hunk 2]
```diff
@@ -0,0 +1,278 @@
...
+One option which should be mentioned separately is
+``credentialsSecret`` which is a `Kubernetes
+secret <https://kubernetes.io/docs/concepts/configuration/secret/>`_
+for backups. Value of this key should be the same as the name used to
+create the secret object (``cluster1-azure-credentials`` in the last
+example).
+
+You can use :ref:`prefix<backup-storages-azure-prefix>` option to specify the
+path (sub-folder) to the backups inside the container. If prefix is not set,
+backups will be stored in the root directory of the container.
```
Comment (egegunes) [inline source/backups.rst:165]: it's not true for this release

**Review by egegunes — CHANGES_REQUESTED** [review]

File: source/conf.py [hunk 3]
```diff
@@ -44,6 +44,7 @@
 ps57recommended = '5.7.35'
 ps80recommended = '8.0.25'
```
Comment (tplavcic) [inline source/conf.py:45]: ps80recommended='8.0.28'
and for ps57 we don't support right now and I don't think we will.

File: source/conf.py [hunk 4]
```diff
@@ -44,6 +44,7 @@
...
 pmm2recommended = '2.25.0'
```
Comment (tplavcic) [inline source/conf.py:46]: pmm2recommended = '2.28.0'

File: source/operator.rst [hunk 5]
```diff
@@ -514,3 +514,320 @@ options for Percona Monitoring and Management.
...
++-----------------+-------------------------------------------------------------------------------------------+
+|                                                                                                             |
++-----------------+-------------------------------------------------------------------------------------------+
+|                 | .. _backup-image:                                                                         |
+|                 |                                                                                           |
+| **Key**         | `backup.image <operator.html#backup-image>`_                                              |
++-----------------+-------------------------------------------------------------------------------------------+
+| **Value**       | string                                                                                    |
++-----------------+-------------------------------------------------------------------------------------------+
+| **Example**     | ``percona/percona-xtrabackup:{{{pxbrecommended}}}``                                             |
```
Comment (tplavcic) [inline source/operator.rst:548]: In this release it will be `percona/percona-server-mysql-operator:0.2.0-backup`, so we will not use official PXB image.

**Review by tplavcic — CHANGES_REQUESTED** [review]

File: source/backups.rst [hunk 6]
```diff
@@ -0,0 +1,278 @@
...
+
+Making on-demand backup
+-----------------------
+
+To make an on-demand backup, the user should first make changes in the 
+``deploy/cr.yaml`` configuration file: set the ``backup.enabled`` key to
+``true`` and configure backup storage in the ``backup.storages`` subsection. 
+
+When the ``deploy/cr.yaml`` file
+contains correctly configured keys and is applied with ``kubectl`` command, use
```
[the remark this answers is not in the document]
Comment (spron-in) [reply source/backups.rst:177]: i'm not saying we should mention helm, i think this is not relevant at all.

**Review by spron-in — CHANGES_REQUESTED** [review]

File: source/assets/images/backup-s3.svg [hunk 7]
```diff
@@ -0,0 +1,524 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
[the remark this answers is not in the document]
Comment (egegunes) [reply source/assets/images/backup-s3.svg:1]: Well... we both have a sidecar to run backup on and a separate backup pod to trigger backup.

File: source/conf.py (hunk 4 above)
Comment (fiowro) [reply source/conf.py:46]: thanks, fixed versions in a separate PR

File: source/backups.rst [hunk 8]
```diff
@@ -0,0 +1,278 @@
...
+      EOF
+
+.. _backups-restore:
+
+Restore the cluster from a previously saved backup
+--------------------------------------------------
+
+Following things are needed to restore a previously saved backup:
+
+* Make sure that the cluster is running.
```
[the remark this answers is not in the document]
Comment (fiowro) [reply source/backups.rst:214]: not yet supported

File: source/backups.rst (hunk 6 above)
[the remark this answers is not in the document]
Comment (fiowro) [reply source/backups.rst:168]: we had a ticket some time ago to mention apply command for all cr.yaml edits... Let's discuss it separately and change the wording for all operators at once

**Review by tplavcic — APPROVED** [review]

**Review by egegunes — APPROVED** [review]

**Pull request merged**